### DISCUSSION TITLE
Common questions

### DISCUSSION THREAD

**Discussion by Yue-Yang** [Yue-Yang/ChatGPT-Siri#30] [Announcements]
# If you can't find the choices:

- First, check the network to make sure you can access the website https://api.openai.com. If you are in China, you may need to use a proxy.
- Second, your API key may be invalid: check your email to see if it has been banned.
- Third, check the OpenAI API dashboard to see if you have enough balance and if your balance has expired. If you are on a free trial, you may need to link a payment method: https://platform.openai.com/account/usage, price: https://openai.com/pricing

Usually, these steps can solve the issue.

# If you cannot start with Siri

first open Settings -> Siri & Search -> Siri Responses ->  Choose "Automatic" and select "Always Show Speech". This will enable you to check if Siri accurately recognizes your speech.

# Can't find file, txt file not exits

You need to open 'Settings' - Apple ID - iCloud - iCloud Drive function to enable the feature of saving to txt file.

# Can't install 1.2

My testing device is iOS 16. According to user feedback, version 1.2 cannot be installed on lower versions of iOS. You may choose to install a lower version. 

# Others

- How to use API key safely: https://help.openai.com/en/articles/5112595-best-practices-for-api-key-safety
- API keys: https://platform.openai.com/account/api-keys
- API usage: https://platform.openai.com/account/usage
- API price: https://openai.com/pricing
- ChatGPT API FAQ: https://help.openai.com/en/articles/7039783-chatgpt-api-faq

**Comment by DoogH2004** [comment]
![image](https://user-images.githubusercontent.com/33561786/224570568-d98b3ab5-da2a-46ba-a021-77b7690daf38.png)

What should I do?

**Reply by Clemdu63** [reply]
got the same issue, can’t work

**Reply by Yue-Yang** [reply]
Your key might be invalid, follow the prompt above to troubleshoot your issue.

**Reply by Clemdu63** [reply]
I generated new keys multiple times and still the same issue

**Reply by Justwords** [reply]
I’m getting the same

**Reply by pierrounew** [reply]
The same for me

**Reply by jonatjin** [reply]
same, even after adding a payment method

**Reply by IgorVinson** [reply]
I see the same, Is this problem solving?

**Reply by IgorVinson** [reply]
I had this problem. I did`t have money on my account. when I paid everything worked

**Reply by matthoganproductions** [reply]
Same issue, I added payment method but it seems there is no way to "pre-pay" and you only pay as you go.  Still having the error after adding a new payment method.

**Reply by JoshB3rry** [reply]
No one figured this out?

**Reply by msllrs** [reply]
still getting the same error even with paid account

**Reply by harupong** [reply]
Try selecting gpt-3.5-turbo for a model. It's highly likely that gpt-4 model is not enabled on your account.

**Reply by matthoganproductions** [reply]
> Same issue, I added payment method but it seems there is no way to "pre-pay" and you only pay as you go. Still having the error after adding a new payment method.

I made a new API key and it started working, not sure what changed but either way it's working now

**Reply by ceti-byte** [reply]
Hallo, ich hab das gleiche Problem. 
![IMG_1047](https://user-images.githubusercontent.com/70188293/229333062-eca198b3-382d-456b-8dce-d73c7fb40da3.jpeg)

Genug Guthaben ist vorhanden. Ich habe schon diverse Keys generiert, leider ohne Erfolg. Hat jmd. eine Idee?

**Comment by mathewben** [comment]


**Reply by Yue-Yang** [reply]
check your balance, https://platform.openai.com/account/usage, Check your balance, it must be greater than 0 and not expired.

**Comment by stormzaba** [comment]
![Uploading 9C118FF3-A20E-4B25-A7E3-BC4C9D7A8352.png…]()

it said text file not exist

**Reply by Yue-Yang** [reply]
You need to open 'Settings' - Apple ID - iCloud - iCloud Drive function to enable the feature of saving to txt file.

**Reply by harrydarwin** [reply]
my drive is on but  theres still no file in notes

**Comment by eternal-learnerdotcom** [comment]
Hello @Yue-Yang, 
Thank you for this shortcut.
I would like to have longer answers from ChatGPT (in number of tokens) and also have the answers printed for a longer time on the screen. 
Unfortunately, I don't see a thing setting in the shortcut's configuration. 
Could you please highlight it for me?
Many thanks. Jon

**Comment by trixlme** [comment]
Great work! I am not figuring how Siri replies are verbalised and that that written answer stays on longer... any ideas?

**Reply by armandrau** [reply]
Click on the text once it should open it as a text file.

**Reply by eternal-learnerdotcom** [reply]
It Doesn't work for me.
By the way, there is a variable called "chatgpt-output-time", but unfortunately, it doesn't account for the answer's time on screen.
Anyway, I don't think this is controllable.
In parameters, you can force Siri to verbose the answers, so it'll take a bit longer until it plays all the messages. 
Cheers

**Reply by eternal-learnerdotcom** [reply]
Hi @armandrau, tried, but unfortunately, nothing happens (at least on the iPhone). But forcing Siri to speak out the content helps a bit...

**Comment by armandrau** [comment]
Brilliant work! Any way to set an automatic first prompt, to jailbreak GPT (such as DAN mode) before you input your first question? That will be super useful.

**Reply by ScrewedDriver** [reply]
How can I know, that ChatGPT is not listening anymore?
I try to end the conversation by voice but it seems to end in a loop.

**Comment by 2902539287** [comment]
只能连续对话6条，超过6条就提示找不到词典键“choices”的值

**Comment by kenschuster** [comment]
I have a GPT-4 subscription, and added the API to the shortcut, but get "Request Error,  invalid_request_error: model_not_found, The model: 'gpt-4' does not exist." What do I need to correct?
![cap 2023-04-09 at 11 26 28 AM](https://user-images.githubusercontent.com/39962080/230782778-907b39bb-d483-424f-b004-32f46e05f5e3.png)

**Comment by mappenfeller** [comment]
When I change the "exit-command" to something other than the default "Quit Chat" text, it seems to stick me in a loop and I can't end the shortcut. Am I doing something wrong? (AND THANK YOU - this is a game changer for Siri on the HomePod!)

**Comment by TheInnerPrince** [comment]
Can you help me with the siri shortcut to stop asking to « continue » when he is done answering ? I’d like him to keep displaying the chatGPT response on the screen :s

**Comment by NazariiDmytryshyn** [comment]
Hello) Great team, I have one question, I installed version 1.2.2 on my mac, but I would like to see the text that Siri reads, at the moment I see only one line like "I finished reading this answer"
How can I do this?
![2023-05-05_11-40](https://user-images.githubusercontent.com/84309391/236413608-0f4adfec-70c1-4dda-9640-6a5c3cb68a2e.png)

**Comment by truynal** [comment]
AI画图怎么用，图总是画不好

**Comment by daiglels** [comment]
Trying to get this working. I get the "file not found" error on my iPhone and iPad running iOS 17. However, I can run the Shortcut on 2 of my Macs without issue. Any thoughts?

**Comment by Huobowei** [comment]
how can i change direction-model from model to deepseek-reasoner
![微信图片_20250314095123](https://github.com/user-attachments/assets/5acc1b11-fa43-4e73-bc49-b4ac9032b5ef)
